runtime/src/identity.rs

```diff
@@ -379,6 +395,42 @@ decl_module! {
             Ok(())
         }
 
+        /// # TODO
+        ///  -
+        fn forwarded_call(origin, target_did: IdentityId, proposal: Box<T::Proposal>) -> Result {
+            let sender = ensure_signed(origin)?;
+
+            // 1. Constraints.
+            // 1.1. A valid current identity.
+            let current_did = <CurrentDid>::get();
+            ensure!( current_did.is_some(), "Missing current identity on the transaction");
+
+            // TODO 1.2. Check that current_did is a signing key of target_did
+            // ensure!(isSigningKey(current_did, target_did))
+
+            // 1.3. Check that target_did has a KYC (this has
+            // already been done for current_did in the SignedExtension)
+            ensure!(Self::has_valid_kyc(target_did), "Invalid KYC validation on target did");
+
+            // 2. Actions
+            <CurrentDid>::put(target_did);
+
+            // Also set current_did roles when acting as a signing key for target_did
```

Comment:
Do you think we need something like `CurrentRoles` similar to `CurrentDid` for this?

Reply:
I don't think so because we won't need to update/change roles of an entity during a transaction. Will we?